dashboard/app/controllers/teacher_feedbacks_controller.rb

```diff
@@ -20,8 +20,24 @@ def index
       )
     end
 
+    # We need to know which feedback is the latest for a level/script in order to determine
+    # if the level is awaiting another review from the teacher. (A level is awaiting review if it was marked
+    # as keep working by the teacher and then the student made progress. This status can only apply to the
+    # latest feedback left by the teacher)
+    #
+    # marked_latest_for_level is a map from level/script to true if we've marked the latest feedback for the level
+    # in our list
+    marked_latest_for_level = {}
+
     @feedbacks_as_student_with_level_info = @feedbacks_as_student.map do |feedback|
-      feedback.attributes.merge(feedback&.get_script_level&.summary_for_feedback)
+      level_feedback_key = [feedback.level_id, feedback.script_id].join('_')
+      is_latest = !marked_latest_for_level[level_feedback_key]
+
+      if is_latest
+        marked_latest_for_level[level_feedback_key] = true
+      end
+
+      feedback.summarize.merge({is_latest_for_level: is_latest}).merge(feedback&.get_script_level&.summary_for_feedback)
```

Comment:
(minor) This line is a little hard to parse, maybe split it up into a few lines?  (No need to do this if you aren't planning on making other changes.)

Reply:
I'll make a note and circle back on this (don't want to wait another hour for drone)